library/scheduler.hh

```diff
@@ -176,10 +181,36 @@ namespace dripline
             f_scheduler_thread()
     {}
 
+    template< typename executor, typename clock >
+    scheduler< executor, clock>::scheduler( scheduler< executor, clock >&& a_orig ) :
+        f_exe_buffer( std::move(a_orig.f_exe_buffer) ),
+        f_cycle_time( std::move(a_orig.f_cycle_time) ),
+        f_the_executor(),
+        f_events( std::move(a_orig.f_events) ),
+        f_scheduler_mutex(),
+        f_executor_mutex(),
+        f_cv(),
```

Comment:
Do these not need to be moved or copied from a_orig?

Reply:
No.  Mutexes and condition variables are explicitly non-copyable and aren't moveable either.  I think you run into trouble when you consider moving synchronization objects like these. 

Effectively, the move operator= is just inherently non-thread-safe, and should never be used while the relevant class is running.